### DISCUSSION TITLE
Вендоринг зависимостей

### DISCUSSION THREAD

**Discussion by lowitea** [dobroweb/rhythmbox-plugin-yandex-music#14] [Ideas]
Для того чтобы не зависеть от установленных зависимостей в систему и не требовать их глобальной установки от пользователя, можно попробовать положить зависимости в сам пакет.

**Comment by feschukov** [comment]
Так в этом и принцип Linux, что пакет содержит только собственный код, а зависимости устанавливаются отдельно и пакет их не тянет. Мне нравится этот принцип, поэтому я не вкладываю в пакет зависимости.

К тому же, `pip`, запущенный от пользователя, устанавливает `yandex-music` в папку `home` пользователя. Плагин также устанавливается в папку `home` пользователя, если делать это по инструкции из `README`. Так что глобально устанавливать зависимости теоретически не требуется.

В принципе, можно делать версию плагина при релизе, которая будет включать `yandex-music` в себе. Но надо это сделать так, чтобы код плагина не требовалось менять.

**Reply by lowitea** [reply]
Да, полностью согласен, но что касается зависимостей питона их лучше не ставить глобально через pip, даже от пользователя, могут возникнуть конфликты зависимостей, если какие-то системные зависимости будут зависеть от тех же библиотек, но других версий, от которых зависит `yandex-music`. В таком случае pip просто перезапишет уже скачанные библиотеки другими версиями, от чего могут возникнуть ошибки совместимости других пакетов установленных пользователем. Если идти по linux way. То лучше тогда собирать пакет .deb/.rpm, в котором будут указаны зависимости из репозиториев дистрибутива. Таким образом можно избежать конфликтов. Но самой `yandex-music` нет в репозиториях популярных дистрибутивов насколько я вижу, поэтому это будет сложно сделать. Вариант включать пакет `yandex-music` при сборке мне кажется отличным, в репозиторий действительно класть его не обязательно. В любом случае это просто предложение, для обеспечения большего удобства при установке.
